tests/pytests/unit/cloud/test_cloud.py

```diff
@@ -0,0 +1,38 @@
+import copy
+
+import pytest
+import salt.config
+from salt.cloud import Cloud
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def master_config():
+    opts = copy.deepcopy(salt.config.DEFAULT_MASTER_OPTS)
+    opts["parallel"] = False
+    opts["providers"] = {
+        "test": {},
+    }
+    return opts
+
+
+@pytest.fixture
+def vm_config():
+    return {
+        "driver": "test",
+        "name": "test",
+        "provider": "test:test",
+    }
+
+
+def test_cloud_create_attempt_sync_after_install(master_config, vm_config):
+    master_config["sync_after_install"] = "all"
+    cloud = Cloud(master_config)
+    cloud.clouds["test.create"] = lambda x: True
+
+    with patch(
+        "salt.client.get_local_client",
+        MagicMock(return_value=MagicMock(return_value=True)),
+    ):
+        ret = cloud.create(vm_config)
+    assert ret
```

Comment:
can we also verify that we called the `saltutil.sync_{}` command? I'm just trying to ensure we also ensure we are hitting the part of the code that we intend and call the correct function.

Reply:
What's the best way to accomplish that? I tried messing around with patching the client, but the cloud code is inside a `with` statement (line 1268) so the `MagicMock` object is closed before it's passed back to the test code. This makes it impossible to check `assert_called`, `assert_called_with`, etc. on return.  @waynew , any test magic tricks?